source/blog/2017-12-08-host-deploy-customization.html.md

```diff
@@ -0,0 +1,89 @@
+which is much more easier and powerfull than the database modification.
+
+## The host-deploy playbook
+Currently when the host is added to the engine or reinstalled, we execute the host-deploy playbook
+located at `/usr/share/ovirt-engine/playbooks/ovirt-host-deploy.yml`.
+
+This playbook executes few internal roles to configure for example OVN driver, libvirt guests or FirewallD.
+
+## Special tasks file
+
+As part of this role we also include additional tasks, which could be written by user, to modify the host-deploy
+process for example to open some more FirewallD ports.
+
+Those additional tasks can be added to following file:
+
+```
+/etc/ovirt-engine/ansible/ovirt-host-deploy-post-tasks.yml
+```
+
+## Example
+An example post-tasks file is provided by `ovirt-engine-tools` RPM, at following location:
+
+```
+/etc/ovirt-engine/ansible/ovirt-host-deploy-post-tasks.yml.example
+```
+
+But to see this whole thing in action, you need to create proper file without the `.example` suffix and proper custom tasks:
+
+```
+$ touch /etc/ovirt-engine/ansible/ovirt-host-deploy-post-tasks.yml
+```
+
+Now let's add some tasks which should be executed on the host.
+
+```bash
+$ cat << EOF >> /etc/ovirt-engine/ansible/ovirt-host-deploy-post-tasks.yml
+> ---
+> - name: Enable custom firewall port
+>   firewalld:
+>     port: "12345/tcp"
+>     permanent: yes
+>     immediate: yes
+>     state: enabled
+>
+> - name: Print debug information
+>   debug:
+>     msg: "My custom tasks was successfully executed!"
+> EOF
+```
+
+Now add the new host, and see the Ansible host-deploy log, if our custom post-tasks was executed,
+you can check it by seeing the following log file:
+
+```
+$ cat /var/log/ovirt-engine/host-deploy/ovirt-host-deploy-ansible-20171208153558-192.168.121.36-cdcdc48.log
+
+...
+<snip>
+...
+
+2017-12-08 15:36:12,001 p=4777 u=omachace |  TASK [Executing post tasks defined by user] ************************************
+2017-12-08 15:36:12,020 p=4777 u=omachace |  included: /etc/ovirt-engine/ansible/ovirt-host-deploy-post-tasks.yml for 192.168.121.36
+2017-12-08 15:36:12,026 p=4777 u=omachace |  TASK [Enable custom firewall port] *********************************************
+2017-12-08 15:36:12,633 p=4777 u=omachace |  changed: [192.168.121.36]
+2017-12-08 15:36:12,639 p=4777 u=omachace |  TASK [Print debug information] *************************************************
+2017-12-08 15:36:12,663 p=4777 u=omachace |  ok: [192.168.121.36] => {
+    "msg": "My custom tasks was successfully executed!"
+}
+2017-12-08 15:36:12,665 p=4777 u=omachace |  PLAY RECAP *********************************************************************
+2017-12-08 15:36:12,665 p=4777 u=omachace |  192.168.121.36             : ok=23   changed=4    unreachable=0    failed=0
+```
+
+As you can see we use same log naming as we used for the otopi based deploy process but we have added `-ansible` suffix to the log file name.
```

Comment:
I'd also mention that post-host-deploy task can be used for any task, not only firewalld customization. And we should also mention that IPTables support is deprecated in 4.2 and will be removed in 4.3

Reply:
done